under rootbld, all supplementary groups are nobody;
however, it's not a valid group for chgrp

diff --git a/tests/test-preserve.sh b/tests/test-preserve.sh
index f3e9b87..e39fb83 100755
--- a/tests/test-preserve.sh
+++ b/tests/test-preserve.sh
@@ -22,7 +22,7 @@ init_dirs
 printf '127.0.0.1\tlocalhost\n' > $hosts
 
 chmod 0600 $hosts
-group=$(groups | tr ' ' '\n' | tail -1)
+group=$(groups | tr ' ' '\n' | grep -v nobody | tail -1)
 chgrp $group $hosts
 
 [ -x /usr/bin/chcon ] && selinux=yes || selinux=no
